qa/L0_http/test.sh

```diff
@@ -667,6 +670,47 @@ set -e
 kill $SERVER_PID
 wait $SERVER_PID
 
+### Test Restricted  APIs ###
+MODELDIR="`pwd`/models"
+SERVER_ARGS="--model-repository=${MODELDIR}
+             --http-restricted-api=model-repository,health:k1=v1 \
+             --http-restricted-api=metadata,health:k2=v2"
```

Comment:
IIUC, we have a test with:
1. two groups, two endpoints per group, but health endpoint appears twice, so failure
2. two groups, one endpoint per group, success

Can you add a test like (1) but with success to test multiple groups sharing same key works OK?
3. two groups, two endpoints per group, all unique endpoints, success

Reply:
added support for tests to match grpc tests - covers two categories with a single header value.

added test for unknown category

all restrictions - deem that out of scope for this PR - but good for future coverage